src/Metalama.Patterns.Contracts/Numeric/DoubleBound.cs

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code.SyntaxBuilders;
+
+namespace Metalama.Patterns.Contracts.Numeric;
+
+#pragma warning disable SA1124
+
+[RunTimeOrCompileTime]
+internal sealed class DoubleBound : NumericBound
+{
+    private readonly double _value;
+
+    public DoubleBound( double value, bool isAllowed ) : base( isAllowed )
+    {
+        this._value = value;
+    }
+
+    public override object ObjectValue => this._value;
+
+    #region Conversions
+
+    internal override bool TryConvertToByte( out byte value, out ConversionResult conversionResult )
+    {
+        switch ( this._value )
+        {
+            case < byte.MinValue:
+                value = 0;
+                conversionResult = ConversionResult.TooSmall;
+
+                return false;
+
+            case > byte.MaxValue:
+                value = 0;
+                conversionResult = ConversionResult.TooLarge;
+
+                return false;
+
+            default:
+                value = (byte) this._value;
+
+                conversionResult = value switch
+                {
+                    byte.MinValue => ConversionResult.ExactlyMinValue,
+                    byte.MaxValue => ConversionResult.ExactlyMaxValue,
```

Comment:
This is going to say that `0.1` is `ExactlyMinValue`, but will only produce `ExactlyMaxValue` for exactly `255`. I don't think that was intended.

Reply:
Yes this is by design.